app/models/order.go

```diff
@@ -62,15 +62,44 @@ func init() {
 		return nil
 	})
 	OrderState.State("paid").Enter(func(value interface{}, tx *gorm.DB) error {
+		var orderItems []OrderItem
+
+		tx.Model(value).Association("OrderItems").Find(&orderItems)
+		for _, item := range orderItems {
+			if err := ItemState.Trigger("pay", &item, tx); err != nil {
+				return err
+			}
+		}
+		tx.Save(value)
 		// freeze stock, change items's state
 		return nil
 	})
 	OrderState.State("paid_cancelled").Enter(func(value interface{}, tx *gorm.DB) error {
 		// do refund, release stock, change items's state
 		return nil
 	})
-	OrderState.State("processing")
-	OrderState.State("shipped")
+	OrderState.State("processing").Enter(func(value interface{}, tx *gorm.DB) error {
+		var orderItems []OrderItem
+		tx.Model(value).Association("OrderItems").Find(&orderItems)
+		for _, item := range orderItems {
+			if err := ItemState.Trigger("process", &item, tx); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+	OrderState.State("shipped").Enter(func(value interface{}, tx *gorm.DB) error {
+		tx.Model(value).UpdateColumn("shipped_at", time.Now())
+
+		var orderItems []OrderItem
+		tx.Model(value).Association("OrderItems").Find(&orderItems)
+		for _, item := range orderItems {
+			if err := ItemState.Trigger("pay", &item, tx); err != nil {
```

Comment:
This should be `ItemState.Trigger("ship", &item, tx)`, right? Looks like a copy'n'paste misstep.